packages/next/build/webpack/plugins/middleware-plugin.ts

```diff
@@ -164,6 +167,13 @@ function getCreateAssets(params: {
         { regexp: namedRegex },
       ]
 
+      const regions =
+        metadata.userConfig &&
+        (Array.isArray(metadata.userConfig.regions) ||
+          typeof metadata.userConfig.regions === 'string')
```

Comment:
**question:** shouldn't this type of validation be done inside get-page-static-info.ts? like we do with `runtime` value?

There is also this [function](https://github.com/Schniz/next.js/blob/8cc7b0859cf5c4cea38a43e27e3ad11b0591ac45/packages/next/build/analysis/get-page-static-info.ts#L177), `getMiddlewareConfig()` which runs validation checks for the content of the exported config. Could be the right place.

We also have specific tests for such validations: https://github.com/vercel/next.js/blob/canary/test/production/edge-config-validations/index.test.ts

Reply:
<picture data-single-emoji=":done:" title=":done:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..1srK4Zd13qAphgLB.riOS9qOSs_aULUlinBrkDFXpEU_4ioO5jx-mCvAmjMFXhXJPXCZ2UFyYTBfyKXXsu8PWw9LrUlPP8-MfnwxIwctdDq360J4EAe7rabThfUTPqqPZEyQ.JkyuiTcoUAP-04AbtIP2tQ" alt=":done:" align="absmiddle"></picture> <picture data-single-emoji=":blob_thanks:" title=":blob_thanks:"><img class="emoji" width="20" height="auto" src="https://single-emoji.vercel.app/api/emoji/eyJhbGciOiJkaXIiLCJlbmMiOiJBMjU2R0NNIn0..2mOQ_hL_R8mfjavk.xdEfB1jLW7jVoUxYH8-F5Lf48ULim3amd5c8mBqk6173J0tHHDChLmY9mBeJVIFVyvwNxUVFukIAklHYqkjiH77BwLOvGZi8OPjDpoZSA20ynOPY8VP0yTB6Pzdo.ltrT089TTlxWPKOGRTdr6A" alt=":blob_thanks:" align="absmiddle"></picture>